core/server/worker/src/test/java/alluxio/worker/block/BlockMasterClientTest.java

```diff
@@ -12,24 +12,109 @@
+import alluxio.grpc.CommitBlockPRequest;
+import alluxio.grpc.CommitBlockPResponse;
+import alluxio.grpc.ConfigProperty;
+import alluxio.grpc.GetRegisterLeasePRequest;
+import alluxio.grpc.GetRegisterLeasePResponse;
+import alluxio.grpc.GetWorkerIdPRequest;
+import alluxio.grpc.GetWorkerIdPResponse;
+import alluxio.grpc.GrpcServer;
+import alluxio.grpc.GrpcServerAddress;
+import alluxio.grpc.GrpcServerBuilder;
+import alluxio.grpc.GrpcService;
+import alluxio.grpc.GrpcUtils;
 import alluxio.grpc.LocationBlockIdListEntry;
+import alluxio.grpc.Metric;
+import alluxio.grpc.RegisterWorkerPRequest;
+import alluxio.grpc.RegisterWorkerPResponse;
+import alluxio.grpc.ServiceType;
+import alluxio.grpc.StorageList;
 import alluxio.master.MasterClientContext;
+import alluxio.retry.RetryUtils;
+import alluxio.security.authentication.AuthType;
+import alluxio.security.user.ServerUserState;
+import alluxio.wire.TieredIdentity;
+import alluxio.wire.WorkerNetAddress;
 
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
 import com.google.common.collect.ImmutableSet;
+import io.grpc.stub.StreamObserver;
+import org.junit.After;
+import org.junit.Rule;
 import org.junit.Test;
 
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.HashMap;
 import java.util.HashSet;
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.stream.Collectors;
 
 public class BlockMasterClientTest {
+
+  // alert: you CANNOT just set TEST_SOCKET_ADDRESS to
+  // configuration, because it will get turned to string "localhost/127.0.0.1:9999"
+  // This string cannot be parsed accurately by the current parsing algorithm
+  private static final InetSocketAddress TEST_SOCKET_ADDRESS =
+      new InetSocketAddress("localhost", 9999);
+  private static final String TEST_SOCKET_ADDRESS_STRING = "localhost:9999";
+
+  @Rule
+  public ConfigurationRule mConfiguration = new ConfigurationRule(ImmutableMap
+      .of(
+          PropertyKey.MASTER_RPC_ADDRESSES, ImmutableList.of(TEST_SOCKET_ADDRESS_STRING),
+          // set retry durations shorter to ensure that tests don't take too long
+          PropertyKey.USER_RPC_RETRY_MAX_DURATION, "5s",
+          PropertyKey.USER_RPC_RETRY_BASE_SLEEP_MS, "100ms",
+          PropertyKey.USER_RPC_RETRY_MAX_SLEEP_MS, "500ms",
+          // we don't serve authorization/configuration grpc requests in testing,
+          // so make sure clients are not supposed to make those requests
+          PropertyKey.SECURITY_AUTHENTICATION_TYPE, AuthType.NOSASL,
+          PropertyKey.USER_CONF_CLUSTER_DEFAULT_ENABLED, false),
+      Configuration.modifiableGlobal());
+
+  // mock rpc server managed by test suite
+  private GrpcServer mServer;
+
+  private final AlluxioConfiguration mConf = Configuration.global();
+
+  @Test
+  public void testClientInfo() {
```

Comment:
btw you don't need the string "test" in the test method name, they are marked as `@Test` already. So this one can be just named `clientInfo()`. This applies to all other test methods.

Reply:
Fixed